api/layer/multipart_upload.go

```diff
@@ -312,7 +312,19 @@ func (n *layer) uploadPart(ctx context.Context, multipartInfo *data.MultipartInf
 		},
 	}
 
-	chunk := n.buffers.Get().(*[]byte)
+	var (
+		chunk          *[]byte
+		isReturnToPool bool
+	)
+
+	if p.Size > n.neoFS.MaxObjectSize() {
```

Comment:
Maybe `n.neoFS.MaxObjectSize()/2`? 63M request allocating below would be worse than the same request reusing 64M buffer.

Reply:
Updated